pika/adapters/select_connection.py

```diff
@@ -937,12 +927,8 @@ def poll(self):
                         self._fd_events[PollEvents.ERROR],
                         self._get_max_wait())
                 else:
-                    # NOTE When called without any FDs, select fails on
-                    # Windows with error 10022, 'An invalid argument was
-                    # supplied'.
-                    time.sleep(self._get_max_wait())
-                    read, write, error = [], [], []
-
```

Comment:
I would prefer to leave this time.sleep() code in place. The reason being that the poller should be able to work without file descriptors. The fact that we presently always have at least one file descriptor is an artifact of the current design. In particular, pushing the running of event loop and implementation of wakeup all the way down to the poller layer is arguably misplaced. As the result, there are several up/down delegations of API calls that don't belong. I am planning to refactor such that running of the event loop and wakeup logic is contained in the IOLoop class, while pollers are only responsible for the polling abstraction, which I believe to be more orthogonal. In that case, poller's implementation won't be guaranteed that it will always be used with >= 1 fd.

Reply:
I don't understand why the `time.sleep` call is there, though. Won't it block everything for 5 seconds, guaranteed?

Reply:
It seems like 5 seconds is too long for a sleep, though, since it will block all fd activity, right?